assets/js/base/components/payment-methods/payment-methods.js

```diff
@@ -64,7 +64,7 @@ const PaymentMethods = () => {
 		...paymentMethodInterface
 	} = usePaymentMethodInterface();
 	const currentPaymentMethodInterface = useRef( paymentMethodInterface );
-	const [ selectedToken, setSelectedToken ] = useState( 0 );
+	const [ selectedToken, setSelectedToken ] = useState( '0' );
```

Comment:
Ideally, yes (and good for you to note). However, there was some dependency for this normalization on the fix I made in here because it was a bit of a chain reaction:

- I fixed the dependency issues in effects.
- That triggered some bugs with what the value of token and sourceId was in various places (because of change in what they'd get set to as code executed).
- Which led to a bunch more logic to keep using numbers for token.
- Which led to me realizing that numbers for the token wasn't a good idea anyways (for reasons mentioned in the pull description).
- Which led to me making the decision to normalize in this pull so that I'm not merging with a fragile fix.

If I could have split this out into its own pull cleanly, I would have.

Reply:
Not surprised! 😅